utilities/cli/README.md

```diff
@@ -0,0 +1,159 @@
+
+# Install packages
+# previous ./run build myapp --install
+./holohub install myapp
+```
+
+### **Container Operations**
+```bash
+# Container build
+# previous ./dev_container build ...
+./holohub build-container --base-img nvidia/holoscan:latest
+
+# Container run
+# previous ./dev_container launch ...
+./holohub run-container --img myapp:dev --ssh-x11 --docker-opts="--entrypoint=bash" --no-docker-build
+
+# Development environment
+# previous ./dev_container vscode ...
+./holohub vscode myapp --language cpp
+```
+
+### **Advanced Features**
+```bash
+# Custom build options
+./holohub build myapp --build-with "op1;op2" --configure-args='-DCUSTOM=ON'
+
+# Benchmarking an application (only for applications/workflows)
+./holohub build myapp --benchmark
+```
+
+###  **Development Environment**
+
+Development commands remain familiar:
+
+```bash
+# Environment setup (unchanged)
+[sudo] ./holohub setup
+
+# Enhanced linting with integrated dependency management
+./holohub lint --install-dependencies  # Previously ./run install_lint_deps
+./holohub lint --fix                   # Same clean interface
+
+# Consistent naming conventions
+./holohub clear-cache                  # Previously ./run clear_cache
+./holohub list                         # Unchanged
+```
+
+## New Capabilities & Enhancements
+
+### **Project Creation and Testing**
+The new CLI introduces development lifecycle features:
+
+```bash
+# Scaffold new projects with templates
+./holohub create my_new_app --language cpp
+
+# Integrated testing framework
+./holohub test myapp
+
+# Show environment information
+./holohub env-info
+```
+
+### **Intelligent Development Experience**
+- **Smart Error Messages**: Automatic project name suggestions for typos and better validation
+- **Modern Argument Parsing**: Consistent interface across all commands with improved help text
+- **Enhanced Container Integration**: Better argument forwarding and container lifecycle management
+
+### **Granular Build Control**
+The new architecture provides precise control over your development workflow:
+- **`--local`**: Explicit local development mode
+- **`--no-local-build`**: Skip application rebuild for faster iteration
+- **`--no-docker-build`**: Use existing container images
+- **Dedicated Commands**: Separate `build`, `run`, `build-container`, and `run-container` commands for clear workflow control
+
+## Getting Help
+
+```bash
+./holohub --help              # General help
+./holohub run --help          # Command-specific help
+./holohub run myapp --verbose --dryrun  # Debug mode
+./holohub list                # Check available projects
+```
+
+
+## Useful Commands
+
+- When adding option that looks like an argument, use `=` instead of whitespace ` `:
+```bash
+--run-args="--verbose"   # instead of --run-args "--verbose"
+```
+
+- To clear unused docker cache during development:
+```
+docker image prune -f && docker buildx prune --filter type=regular -f
+```
```

Comment:
Suggest referencing Docker docs in case recommended approach changes in the future:
```suggestion
- To clear unused docker cache during development:
  - Clear Docker images ([doc](https://docs.docker.com/reference/cli/docker/image/prune/)): `docker image prune`
  - Clear Docker buildx cache ([doc](https://docs.docker.com/reference/cli/docker/buildx/prune/)): `docker buildx prune`
  - Clear Docker containers, networks, images ([doc](https://docs.docker.com/reference/cli/docker/system/prune/): `docker system prune`
````

Reply:
thanks, I was trying to document this specific command, but I think providing the generic pruning docs is also ok